tests/test_model.py

```diff
@@ -0,0 +1,189 @@
+    return '{\n  "model_type": "TestModel"\n}'
+
+
+@pytest.fixture
+def weights_as_json() -> str:
+    return '{\n  "param": [\n    0,\n    0,\n    0\n  ]\n}'
+
+
+@pytest.mark.test_hf_access
+def test_hf_access():
+    assert hf_has_write_access
+
+
+def test_save_pretrained(model, tmp_path, config_as_json, weights_as_json) -> None:
+    model.save_pretrained(tmp_path)
+    assert os.path.exists(tmp_path / model.config_name)
+    with open(tmp_path / model.config_name) as f:
+        assert f.read() == config_as_json
+    assert os.path.exists(tmp_path / model.weights_file_name)
+    with open(tmp_path / model.weights_file_name) as f:
+        assert f.read() == weights_as_json
+
+
+@pytest.mark.skipif(not hf_has_write_access, reason=HF_NO_ACCESS_MSG)
+def test_save_pretrained_push_to_hub(model, caplog, tmp_path, config_as_json, weights_as_json):
+    try:
+        model.save_pretrained(save_directory=tmp_path, push_to_hub=True, repo_id=HF_WRITE_PATH)
+
+        assert tmp_path.joinpath(model.config_name).exists()
+        with open(tmp_path.joinpath(model.config_name)) as f:
+            file_contents = f.read()
+        assert file_contents == config_as_json
+
+        assert tmp_path.joinpath(model.weights_file_name).exists()
+        with open(tmp_path.joinpath(model.weights_file_name)) as f:
+            file_contents = f.read()
+        assert file_contents == weights_as_json
+
+    finally:
+        hf_api.delete_file(model.config_name, HF_WRITE_PATH)
+        hf_api.delete_file(model.weights_file_name, HF_WRITE_PATH)
+
+
+@pytest.mark.skipif(not hf_has_write_access, reason=HF_NO_ACCESS_MSG)
+def test_save_pretrained_push_to_hub_no_repo_id(
+    model, caplog, tmp_path, config_as_json, weights_as_json
+):
+    #   If no repo_id was provided, save_directory will be used instead.
+    # We need to name folder exactly as the Repository Name.
+    #   Token used for this tests should be issued by Repo owner, not anyone else with access,
+    # since repo will be looked up/created under current User's name.
+    #   In this case path also contains Username, which is not necessary, it will be ignored;
+    # But this is a simple solution to make sure folder has the right name if we change repos.
+    path = tmp_path.joinpath(HF_WRITE_PATH)
+    path.mkdir(parents=True, exist_ok=True)
+
+    try:
+        model.save_pretrained(save_directory=path, push_to_hub=True)
+
+        assert path.joinpath(model.config_name).exists()
+        with open(path.joinpath(model.config_name)) as f:
+            file_contents = f.read()
+        assert file_contents == config_as_json
+
+        assert path.joinpath(model.weights_file_name).exists()
+        with open(path.joinpath(model.weights_file_name)) as f:
+            file_contents = f.read()
+        assert file_contents == weights_as_json
+
+    finally:
+        hf_api.delete_file(model.config_name, HF_WRITE_PATH)
+        hf_api.delete_file(model.weights_file_name, HF_WRITE_PATH)
```

Comment:
I think we do not need this, should be already covered by `test_push_to_hub` below + `test_save_pretrained_push_to_hub_no_repo_id` in `test_hf_hub_mixin.py`. Or did you have a special reason in mind?